### ISSUE TITLE
Ambiguous parse of lone `v` parameter

### ISSUE THREAD

**Issue by DavidBuchanan314** [P-H-C/phc-string-format#8]
Under the current specification, parsing a string like `$myhash$v=123` would be ambiguous. Is this `<version>`=123, or a param with name `v`?

An easy way to avoid this ambiguity would be to forbid parameters named `v`. I have no idea if this would conflict with any existing implementations (is Argon2 the only "proper" user of the phc-string-format?)

(I'm not sure why version couldn't have just been another param, but I suppose that ship has long sailed now, heh)

**Comment by DavidBuchanan314** [comment]
I just found another reference to this problem, but it doesn't look like it was ever resolved https://github.com/P-H-C/phc-string-format/pull/4#discussion_r552063721

(for what it's worth, the parser I just wrote by following the spec does the opposite of what simonepri's does - oops!)

By the way, OpenSSL has recently(ish) shipped an Argon2 implementation, but it does *not* include phc-string encode/decode logic. Hence, there's probably renewed interest in people implementing this logic themselves (which is what I'm doing, potentially for the pyca/cryptography bindings)

**Comment by tarcieri** [comment]
> I'm not sure why version couldn't have just been another param, but I suppose that ship has long sailed now, heh

Yes, unfortunately #4 was added due to real-world usage in Argon2 implementations which were technically not following this specification. 

Forbidding `v` as a parameter name makes the most sense in terms of making it unambiguous. Would you like to open a PR to do so?

**Comment by DavidBuchanan314** [comment]
Thanks, just opened a PR
